src/lib.rs

```diff
@@ -1269,6 +1269,18 @@ impl From<key::Hash> for BlockId {
 
 #[wasm_bindgen]
 impl BlockId {
+    pub fn calculate(bytes: &[u8]) -> Hash {
```

Comment:
This calculate a `BlockId` from any random byte slice, so the type safety is very low; it should be the other way around, have a `Block::id(Block) -> BlockId`